sycl/include/sycl/accessor.hpp

```diff
@@ -1149,7 +1146,10 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   // __init variant used by the device compiler for ESIMD kernels.
   // TODO In ESIMD accessors usage is limited for now - access range, mem
   // range and offset are not supported.
-  void __init_esimd(ConcreteASPtrType Ptr) { MData = Ptr; }
+  void __init_esimd(ConcreteASPtrType Ptr) {
+    MData = Ptr;
+    detail::loop<AdjustedDim>([&, this](size_t I) { getMemoryRange()[I] = 0; });
```

Comment:
It looks like previously we didn't zero-init the memory for accessors, why do we need to do it now? It looks like this is not hit for local accessors, so I don't understand.

Reply:
I tried to be minimalistic and initialize only what was used.

operator[] calls getLinearIndex(), which calls getMemoryRange()
That is why I added getMemoryRange() to white-list and inialized MemoryRange field for global accessors passed to ESIMD.
 
I did not add zero-initialization for other fields assuming they were not used for operator[],
but that seems to be wrong after (2nd check): getTotalOffset() reads impl.Offset directly, not via getOffset().

So, yes, I will add initialization for other fields as well (i.g. O.

Hopefully, we can remove __init_esimd() a bit later and in a separate PR
as it requires some extra caution
(or lots of extra caution - now when we see accessor, we simply convert it's MData to surface_index and do not consider adding any extra offsets).

Reply:
ah ok so this part is not for local accessors but for global accessors with operator[], that helps thx.

is there a reason we need to zero init for esimd and not for sycl?